plugins/onepad/onepad.cpp

```diff
@@ -258,8 +259,9 @@ PADfreeze(int mode, freezeData *data)
 
         // Tales of the Abyss - pad fix
         // - restore data for both ports
-        for (int port = 0; port < 2; port++) {
-            for (int slot = 0; slot < 4; slot++) {
+		//more port < 2; to GAMEPAD_NUMBER, change slot for < 4 to < 6?
+        for (int port = 0; port < GAMEPAD_NUMBER; port++) {
+            for (int slot = 0; slot < 6; slot++) {
```

Comment:
I think `port` and `slot` should remain as 2 and 4 everywhere - A PS2 has 2 pad ports and you can plug in a multitap with 4 slots in each port.

Reply:
Very nice catch, I'll add it and compile and test, thanks, I believe I fixed the clang build issue also.

I will double check all port/slot vars

Reply:
The Travis CI clang build also checks the file formatting (which is why the build failed). If you run clang-format on the file (I think `clang-format -i /path/to/file`) it should auto-format it for you.

However, I do spot some weird formatting in the file (the indentation is wrong on some of the lines that you haven't touched), so if it still fails after that then don't worry about it for now.

Reply:
I usually fix programs in java, C isn't my strong suit.  But I'm very good with Linux, so I apologize in advance if I make some rookie mistakes :)

If you have advise on a decent/free IDE to use to make things easier to spot errors I'm more than willing to use it.  I've been direct editing my fork which in hind sight is not the best approach lol.